### PR TITLE
add transformStateWithClones task solution

### PR DESCRIPTION
**Pull request by opiquem**
add transformStateWithClones task solution

### CODE REVIEW

File: src/transformStateWithClones.js [hunk 1]
```diff
@@ -7,7 +7,37 @@
...
-  // write code here
+  const stateClone = { ...state };
+  const transformedState = [];
+
+  for (const action of actions) {
+    switch (action.type) {
+      case 'addProperties':
+        for (const data in action.extraData) {
+          stateClone[data] = action.extraData[data];
+        }
```
Comment (Yuliia-kucheriavenko) [inline src/transformStateWithClones.js:18]: The second option in one line*)) Object.assign(stateClone, action.extraData)

**Review by Yuliia-kucheriavenko — APPROVED** [review]
Well done!

File: src/transformStateWithClones.js (hunk 1 above)
Comment (33Orange) [inline src/transformStateWithClones.js:18]: Let's change this part of the code to `Object.assign`

**Review by 33Orange — CHANGES_REQUESTED** [review]
Good job.

Just one moment:
- Let's change addProperties function to code with Object.assign (Checklist p1).

**Review by Anna-Tkachenko — APPROVED** [review]
⭐️

**Pull request closed without merging**